### PR TITLE
Extract memecaptain API bits into their own library file

### PR DESCRIPTION
**Pull request by lukewaite**
I have my own Hubot currently pointed at my fork of the `hubot-meme` project and can confirm that these changes work.

Extracting the API bits to their own library file allows you to require in the memeGenerator for use in custom in-house memes.

Fixes https://github.com/cycomachead/hubot-meme/issues/8

### CODE REVIEW

**Comment by cycomachead** [comment]
Thank you! I'll take a look at this soon, today hopefully.

File: src/lib/memecaptain.coffee [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        msg.http(res.headers.location).get() (err, res, body) ->
+          return msg.reply "#{baseError} #{err}" if err
+          return if res.statusCode == 200 # wait for the image
+          if res.statusCode == 303
+            msg.send res.headers.location
+            clearInterval(timer)
+          else
+            msg.reply "#{baseError} #{reasonError} #{res.statusCode} while waiting for the image"
+      , 2000)
+    if res.statusCode > 300 # memecaptian error
```
Comment (giftig) [inline src/lib/memecaptain.coffee:64]: typoed memecaptain :stuck_out_tongue:

File: src/lib/memecaptain.coffee [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        height_pct: 0.25
+      }
+    ]
+  }
+  return JSON.stringify(data)
+
+module.exports = (msg, imageID, upperText, lowerText) ->
+  MEME_CAPTAIN = 'http://memecaptain.com/gend_images'
+  baseError = 'Sorry, I couldn\'t generate that meme.'
+  reasonError = 'Unexpected status from memecaptian.com:'
```
Comment (giftig) [inline src/lib/memecaptain.coffee:46]: Aww typoed captain in the URL :disappointed:

**Comment by giftig** [comment]
I just resolved the conflicts with my commits which were recently merged and tried it and it seems to work, so looks good to me :+1:. Those typos were already there, it seems. Sad times.

**Comment by cycomachead** [comment]
Yeah, they got fixed at one point. I'll correct them.

**Comment by cycomachead** [comment]
I've just pushed 1.2 to npm! Thanks @lukewaite and @giftig For the contributions!

**Pull request merged**